make/common/MakeBase.gmk

```diff
@@ -306,16 +306,27 @@ endef
 # There are two versions, either creating a relative or an absolute link. Be
 # careful when using this on Windows since the symlink created is only valid in
 # the unix emulation environment.
+# In msys2 we use mklink /J because its ln would perform a deep copy of the target.
+# This inhibits performance and can lead to issues with long paths. With mklink /J
+# relative linking does not work, so we handle the link as absolute path.
 define link-file-relative
 	$(call MakeTargetDir)
 	$(RM) '$(call DecodeSpace, $@)'
-	$(LN) -s '$(call DecodeSpace, $(call RelativePath, $<, $(@D)))' '$(call DecodeSpace, $@)'
+	if [ "$(OPENJDK_BUILD_OS_ENV)" = "windows.msys2" ]; then \
+	  cmd //c "mklink /J $(call FixPath, $(call DecodeSpace, $@)) $(call FixPath, $(call DecodeSpace, $<))"; \
+	else \
+	  $(LN) -s '$(call DecodeSpace, $(call RelativePath, $<, $(@D)))' '$(call DecodeSpace, $@)'; \
+	fi
```

Comment:
I think it would be better to evaluate this conditional in make rather than the shell. Something like this:
```suggestion
	ifeq ($(OPENJDK_BUILD_OS_ENV), windows.msys2)
	  cmd //c "mklink /J $(call FixPath, $(call DecodeSpace, $@)) $(call FixPath, $(call DecodeSpace, $<))"
	else
	  $(LN) -s '$(call DecodeSpace, $(call RelativePath, $<, $(@D)))' '$(call DecodeSpace, $@)'
	endif
```
Just note that the make conditional lines need to be space indented. Same goes for the expression below.

Reply:
Thanks, that's the kind of suggestions I was hoping for from the build experts. 😄 I pushed a commit, let's see if it works...

Reply:
> Thanks, that's the kind of suggestions I was hoping for from the build experts. 😄 I pushed a commit, let's see if it works...

Hm, obviously not... I guess there must be some indentation error still...